src/Service/Indexer/Enricher/SiteKitSchema2x/PersonDocumentEnricher.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\CityGov\Service\Indexer\Enricher\SiteKitSchema2x;
+
+use Atoolo\Resource\Resource;
+use Atoolo\Resource\ResourceLoader;
+use Atoolo\Search\Service\Indexer\DocumentEnricher;
+use Atoolo\Search\Service\Indexer\IndexDocument;
+use Atoolo\Search\Service\Indexer\IndexSchema2xDocument;
+
+/**
+ * TODO
+ * - person.google.noIndex -> https://gitlab.sitepark.com/sitekit/citygov-php/-/blob/develop/php/SP/CityGov/Component/Initialization.php?ref_type=heads#L218
+ * @phpstan-type Membership array{
+ *      primary?: bool,
+ *      organisation?:array{
+ *          url?:string
+ *      }
+ *  }
+ * @phpstan-type Competence array{
+ *       primary?: bool,
+ *       product?:array{
+ *           url?:string
+ *       }
+ *   }
+ *
+ * @implements DocumentEnricher<IndexSchema2xDocument>
+ */
+class PersonDocumentEnricher implements DocumentEnricher
+{
+    public function __construct(
+        private readonly ResourceLoader $resourceLoader,
+        private readonly OrganisationDocumentEnricher $organisationEnricher
+    ) {
+    }
+
+    public function isIndexable(Resource $resource): bool
+    {
+        return true;
+    }
+
+    public function enrichDocument(
+        Resource $resource,
+        IndexDocument $doc,
+        string $processId
+    ): IndexDocument {
+
+        if ($resource->getObjectType() !== 'citygovPerson') {
+            return $doc;
+        }
+
+        return $this->enrichDocumentForPerson($resource, $doc);
+    }
+
+    private function enrichDocumentForPerson(
+        Resource $resource,
+        IndexSchema2xDocument $doc
+    ): IndexSchema2xDocument {
+
+        $firstname = $resource->getData()->getString(
+            'metadata.citygovPerson.firstname'
+        );
+        $lastname = $resource->getData()->getString(
+            'metadata.citygovPerson.lastname'
+        );
+
+        $doc->sp_citygov_firstname = $firstname;
+        $doc->sp_citygov_lastname = $lastname;
+        $doc->sp_sortvalue = $lastname . 'aaa' . $firstname;
+        if ($lastname !== '') {
+            $doc->sp_citygov_startletter = mb_substr($lastname, 0, 1);
+        }
+
+        /** @var Membership[] $membershipList */
+        $membershipList = $resource->getData()->getArray(
+            'metadata.citygovPerson.membershipList.items'
+        );
+        /** @var array<array<string>> $organisationNameMergeList */
+        $organisationNameMergeList = [];
+        /** @var string[] $organisationTokenList */
+        $organisationTokenList = [];
+
+        foreach ($membershipList as $membership) {
+            $organisationLocation = $membership['organisation']['url'] ?? null;
+            if ($organisationLocation === null) {
+                continue;
+            }
+
+            $organisationResource = $this->resourceLoader->load(
+                $organisationLocation
+            );
```

Comment:
Reacting to errors in the enricher is not helpful, as the document is always indexed incompletely. All errors are intercepted by the indexer per document. If an error occurs for a document, the error is logged and the document is not indexed.

Reply:
Then I believe it would have to declare that/what it can throw. Idealy thats a `DocumentEnrichingException` of some sorts so an implementing source can catch specifically those from any `DocumentEnricher` without having to know what implementations are or could be configured and what those may throw.

Reply:
Is that what you mean?
```php
    /**
     * @throws DocumentEnrichingException
     */
    public function enrichDocument(
        Resource $resource,
        IndexDocument $doc,
        string $processId
    ): IndexDocument {
        ...
    }
```

Where the enricher is called, however, I would catch all errors in any case, as I would not rely on all enricher catching all error cases.

```php
            try {
                $doc = $update->createDocument();
                foreach ($this->documentEnricherList as $enricher) {
                    $doc = $enricher->enrichDocument(
                        $resource,
                        $doc,
                        $processId
                    );
                }
                $documents[] = $doc;
            } catch (Throwable $e) {
                $this->indexerProgressHandler->error($e);
            }
```

This ensures that the indexer runs stably even if an unexpected problem occurs in individual cases. The errors are logged.

Reply:
Yes, that is what I mean.
It would have to be on the interface and every implementation that could throw.

Catching everything should be fine.

Reply:
added this